### PR TITLE
add basic test cases for catalog app

### PR DESCRIPTION
**Pull request by jiaqiluo**
automation is passed

### CODE REVIEW

File: tests/v3_api/test_upgrade.py [hunk 1]
```diff
@@ -573,3 +591,52 @@ def upgrade_rancher_server(serverIp,
...
+    )
+    validate_catalog_app(app.name)
+
+
+def modify_catalog_app(app_name):
+    p_client = namespace["p_client"]
+    app = wait_for_app_to_active(p_client, app_name)
+    externalId = "catalog://?catalog=library&template=mysql&version=0.3.8"
+    p_client.action(obj=app, action_name="upgrade", externalId=externalId)
+    validate_catalog_app(app_name)
```
Comment (sangeethah) [inline tests/v3_api/test_upgrade.py]: @jiaqiluo Could you add some more checks (may be image name if that changes between 2 versions) on the upgraded app other than status being "active" ?

**Comment by jiaqiluo** [comment]
the automation test is passed

File: tests/v3_api/test_upgrade.py (hunk 1 above)
Comment (jiaqiluo) [reply tests/v3_api/test_upgrade.py]: done

**Comment by jiaqiluo** [comment]
the automation test is passed

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**